lib/compute/agent-node-config.ts

```diff
@@ -190,4 +193,61 @@ export class AgentNodeConfig {
        useEphemeralDevices: false,
      };
    }
+
+   private getMacTemplate(stack: Stack, props: AgentNodeNetworkProps): { [x: string]: any; } {
+     return {
+       ami: 'ami-0379811a08268a97e',
+       amiType:
+        { macData: { sshPort: '22' } },
+       associatePublicIp: false,
+       connectBySSHProcess: false,
+       connectionStrategy: 'PRIVATE_IP',
+       customDeviceMapping: '/dev/sda1=:300:true:gp3::encrypted',
+       deleteRootOnTermination: true,
+       description: 'jenkinsAgentNode-Jenkins-Agent-Mac-M1',
+       ebsEncryptRootVolume: 'ENCRYPTED',
+       ebsOptimized: true,
+       hostKeyVerificationStrategy: 'OFF',
+       iamInstanceProfile: this.AgentNodeInstanceProfileArn,
+       labelString: 'Jenkins-Agent-Mac-M1',
```

Comment:
Lets follow the same format as other agent:
```
Jenkins-Agent-MacOS-x64-Mac1Metal-Multi-Host
```

Reply:
I don't think `Multi-Host` is needed

Reply:
@gaiksaya following the existing naming convention we are using `Multi-Host` for more than 1 executors.